web/modules/custom/dpl_dashboard/src/Controller/DplDashboardController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\dpl_dashboard\Controller;
+
+use Drupal\Core\Block\BlockManagerInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Render\RendererInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Render intermediate list react app.
+ */
+class DplDashboardController extends ControllerBase {
+
+  /**
+   * DplDashboardController constructor.
+   *
+   * @param \Drupal\Core\Block\BlockManagerInterface $blockManager
+   *   Drupal block manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupal renderer service.
+   */
+  public function __construct(
+    private BlockManagerInterface $blockManager,
+    private RendererInterface $renderer
+  ) {
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.block'),
+      $container->get('renderer'),
+    );
+  }
+
+  /**
+   * Demo react rendering.
+   *
+   * @return mixed[]
+   *   Render array.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function list(): array {
+    /** @var \Drupal\dpl_dashboard\Plugin\Block\DashboardBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_dashboard_block', []);
+
+    // Some blocks might implement access check.
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```

Comment:
Optionally add a TODO about using a service or other common Drupal permission functionality in the future

Reply:
Ticket create about this, so we remember this optimization: https://platform.dandigbib.org/issues/5801